src/Playwright/Core/RouteHandler.cs

```diff
@@ -24,35 +24,37 @@
 
 using System;
 using System.Text.RegularExpressions;
+using System.Threading.Tasks;
 
 namespace Microsoft.Playwright.Core
 {
-    internal class RouteSetting
+    internal class RouteHandler
     {
         public Regex Regex { get; set; }
 
         public Func<string, bool> Function { get; set; }
 
-        public Action<IRoute> Handler { get; set; }
+        public object Handler { get; set; }
```

Comment:
You should be able to use a `Delegate` here, so you don't need to cast later on:
```suggestion
        public Delegate Handler { get; set; }
```

Reply:
yeah, thanks.